Source/WebCore/rendering/RenderBox.cpp

```diff
@@ -3563,6 +3563,16 @@ void RenderBox::computePreferredLogicalWidths(const Length& minWidth, const Leng
 {
     if (shouldComputeLogicalHeightFromAspectRatio()) {
```

Comment:
Sorry, I didn't express clearly! I mean, if `style().logicalWidth().isFixed()`, there is no need to check aspect-ratio.
It will be something like this:
```
    if (!style().logicalWidth().isFixed() && shouldComputeLogicalHeightFromAspectRatio()) {
        auto [logicalMinWidth, logicalMaxWidth] = computeMinMaxLogicalWidthFromAspectRatio();
        logicalMinWidth = std::max(logicalMinWidth - borderAndPadding, 0_lu);
        logicalMaxWidth = std::max(logicalMaxWidth - borderAndPadding, 0_lu);
        m_minPreferredLogicalWidth = std::clamp(m_minPreferredLogicalWidth, logicalMinWidth, logicalMaxWidth);
        m_maxPreferredLogicalWidth = std::clamp(m_maxPreferredLogicalWidth, logicalMinWidth, logicalMaxWidth);
    }
```

Reply:
Is there a reason we don't want to check the aspect-ratio in that scenario? Would that not be this part of the spec: 

"This transferred minimum is capped by any definite [preferred](https://drafts.csswg.org/css-sizing-3/#preferred-size) or [maximum size](https://drafts.csswg.org/css-sizing-3/#max-width) in the destination axis." 

I believe the definite preferred size here would correspond to style().logicalWidth().isFixed() being true right? Those caps to the definite size would only occur in that scenario?

Reply:
IIUC, in `RenderBlock::computePreferredLogicalWidths()`:
```
m_minPreferredLogicalWidth = m_maxPreferredLogicalWidth = adjustContentBoxLogicalWidthForBoxSizing(lengthToUse);
```
It sets the fixed logicalWidth  value to `m_minPreferredLogicalWidth` and `m_maxPreferredLogicalWidth`, then it calls `RenderBox::computePreferredLogicalWidths`.
Which means, here, the constraint of `logicalWidth` has no affection. `m_minPreferredLogicalWidth` and `m_maxPreferredLogicalWidth` already have the value of  `logicalWidth`, and transferred min/max width are capped/floored by `logicalWidth`, they won't change the value of `m_minPreferredLogicalWidth` and `m_maxPreferredLogicalWidth` anyway.

Reply:
This should be fixed now!

Reply:
Thanks, the patch looks good to me:)